xgcm/test/test_metrics.py

```diff
@@ -44,3 +70,273 @@ def test_metrics_2d_grid():
+        np.ones([len(xt), len(yt)]) * dy + 0.2, coords=[("xt", xt), ("yu", yu)]
+    )
+    dy_e = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dy + 0.3, coords=[("xu", xu), ("yt", yt)]
+    )
+    dy_t = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dy + 0.4, coords=[("xt", xt), ("yt", yt)]
+    )
+
+    dz_t = xr.DataArray(
+        data_generator() * dz, coords=[("xt", xt), ("yt", yt), ("time", t), ("zt", zt)]
+    )
+    dz_w = xr.DataArray(
+        data_generator() * dz, coords=[("xt", xt), ("yt", yt), ("time", t), ("zw", zw)]
+    )
+
+    # Make sure the areas are not just the product of x and y distances
+    area_ne = (dx_ne * dy_ne) + 0.1
+    area_n = (dx_n * dy_n) + 0.2
+    area_e = (dx_e * dy_e) + 0.3
+    area_t = (dx_t * dy_t) + 0.4
+
+    # calculate volumes, but again add small differences.
+    volume_t = (dx_t * dy_t * dz_t) + 0.25
+
+    def _add_metrics(obj):
+        obj = obj.copy()
+        for name, data in zip(
+            [
+                "dx_ne",
+                "dx_n",
+                "dx_e",
+                "dx_t",
+                "dy_ne",
+                "dy_n",
+                "dy_e",
+                "dy_t",
+                "dz_t",
+                "dz_w",
+                "area_ne",
+                "area_n",
+                "area_e",
+                "area_t",
+                "volume_t",
+            ],
+            [
+                dx_ne,
+                dx_n,
+                dx_e,
+                dx_t,
+                dy_ne,
+                dy_n,
+                dy_e,
+                dy_t,
+                dz_t,
+                dz_w,
+                area_ne,
+                area_n,
+                area_e,
+                area_t,
+                volume_t,
+            ],
+        ):
+            obj.coords[name] = data
+            obj.coords[name].attrs["tracked_name"] = name
+        # add xgcm attrs
+        for ii in ["xu", "xt"]:
+            obj[ii].attrs["axis"] = "X"
+        for ii in ["yu", "yt"]:
+            obj[ii].attrs["axis"] = "Y"
+        for ii in ["zt", "zw"]:
+            obj[ii].attrs["axis"] = "Z"
+        for ii in ["time"]:
+            obj[ii].attrs["axis"] = "T"
+        for ii in ["xu", "yu", "zw"]:
+            obj[ii].attrs["c_grid_axis_shift"] = 0.5
+        return obj
+
+    coords = {
+        "X": {"center": "xt", "right": "xu"},
+        "Y": {"center": "yt", "right": "yu"},
+        "Z": {"center": "zt", "right": "zw"},
+    }
+
+    metrics = {
+        ("X",): ["dx_t", "dx_n", "dx_e", "dx_ne"],
+        ("Y",): ["dy_t", "dy_n", "dy_e", "dy_ne"],
+        ("Z",): ["dz_t", "dz_w"],
+        ("X", "Y"): ["area_t", "area_n", "area_e", "area_ne"],
+        ("X", "Y", "Z"): ["volume_t"],
+    }
+    # Used previously for some fail test. Retain for now.
+    # coords_outer = {
+    #     "X": {"center": "xt", "outer": "xu"},
+    #     "Y": {"center": "yt", "outer": "yu"},
+    #     "Z": {"center": "zt", "outer": "zw"},
+    # }
```

Comment:
I would like to remove / resolve all of these commented lines before merging. Can you recommend what to cut / keep?

Reply:
Go ahead and remove. These were just copied over from the prototype I built [here](https://github.com/jbusecke/xarrayutils/blob/master/xarrayutils/test/test_xgcm_utils.py).
I can easily restore them, once I migrate the weighted mean to xgcm proper.